src/odemis/acq/__init__.py

```diff
@@ -315,6 +321,15 @@ def run(self):
                 if not isinstance(das, collections.Iterable):
                     logging.warning("Future of %s didn't return a list of DataArrays, but %s", s, das)
                     das = []
+
+                # Add extra settings to metadata
+                if self._settings_obs:
+                    settings = self._settings_obs.get_all_settings()
+                    for da in das:
+                        da.metadata[model.MD_EXTRA_SETTINGS] = copy.deepcopy(settings)
+                else:
+                    logging.warning("Acquisition task for stream %s has no metadata manager " % s,
```

Comment:
This message needs to be updated. Anyway, it's now the same for all the streams, so probably should just be put at the beginning (or the end), only once.